Lesson 2/src/Main.java

```diff
@@ -0,0 +1,44 @@
+public class Main {
+    private static final int ARRAY_SIZE = 10000;
+
+    public static void main(String[] args) {
+        bubbleSort(generateArr(ARRAY_SIZE));
+        selectSort(generateArr(ARRAY_SIZE));
+        insertSort(generateArr(ARRAY_SIZE));
+    }
+
+    public static ArrayImpl<Integer> generateArr(int size) {
+        ArrayImpl<Integer> array = new ArrayImpl<>(size);
+        for (int i = 0; i < size; i++) {
+            array.add((int) (Math.random() * 1000));
+        }
+        return array;
+    }
+
+    public static void bubbleSort(ArrayImpl<Integer> array) {
+        System.out.println(array);
+        long start = System.nanoTime();
+        array.bubbleSort();
+        long finish = System.nanoTime();
+        System.out.println(array);
+        System.out.printf("Operation time: %d ms \n", finish - start);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.